data-prepper-plugins/kafka-plugins/src/test/java/org/opensearch/dataprepper/plugins/kafka/consumer/KafkaCustomConsumerTest.java

```
@@ -272,6 +272,41 @@ public void testBufferOverflowPauseResume() throws InterruptedException, Excepti
             Assertions.assertNotNull(event.getEventHandle().getExternalOriginationTime());
         }
     }
+
+    @Test
+    public void testKafkaMetadata() {
+        String topic = topicConfig.getName();
+        consumerRecords = createPlainTextRecords(topic, 0L);
+        when(kafkaConsumer.poll(any(Duration.class))).thenReturn(consumerRecords);
+        consumer = createObjectUnderTest("plaintext", false);
+
+        try {
+            consumer.onPartitionsAssigned(List.of(new TopicPartition(topic, testPartition)));
+            consumer.consumeRecords();
+        } catch (Exception e){}
```

Are you expecting these to fail? If so, you can wrap these with `assertThrows`.

e.g.
```
assertThrows(Exception.class, () -> consumer.consumeRecords());
```

If you are not expecting this, why suppress it? Should the test fail? I tend to think to let the exception bubble up.

---

No, not expecting this to throw. I agree with you, it would be better to let the exception bubble out. I was just basing this off of the other tests ithat all swallow exceptions from `consumeRecords()`. I assumed that there was some reason for that, so I followed suit. Since that method `throws Exception`, I've carried that forward on my test, so exceptions will no longer be swallowed.